src/rgw/rgw_rados.cc

```diff
@@ -482,11 +487,9 @@ int RGWZoneGroup::read_default_id(string& default_id, bool old_format)
     /* try using default realm */
     RGWRealm realm;
     int ret = realm.init(cct, store);
-    if (ret < 0) {
-      ldout(cct, 10) << "could not read realm id: " << cpp_strerror(-ret) << dendl;
-      return -ENOENT;
+    if (ret == 0) {
+      realm_id = realm.get_id();
```

Comment:
actually, it doesn't quite right , maybe the right way should like : 
if (ret == 0) {
realm_id = realm.get_id();
}else if( ret != -ENOENT){
 ldout(cct, 10) << "could not read realm id: " << cpp_strerror(-ret) << dendl;
 return ret;
}

Thank you for reply! @cbodley